File: app/src/main/java/com/blackmirror/dongda/utils/DensityUtils.java
package com.blackmirror.dongda.utils;

import android.util.DisplayMetrics;

/**
 * 像素Dp Px相关
 * Created by Ruge on 2018-05-03 下午4:46
 */
public class DensityUtils {

    public static float px2dp(float pxVal) {
        final float scale = AYApplication.appContext.getResources().getDisplayMetrics().density;
        return (pxVal / scale);
    }

    /**
     * dp转px
     *
     * @param dp
     * @return
     */
    public static int dp2px(int dp) {
        float scale = AYApplication.appContext.getResources().getDisplayMetrics().density;
        return (int) (dp * scale + 0.5f);
    }

    public static int getScreenWidthDp() {
        //2、通过Resources获取
        DisplayMetrics dm = AYApplication.appContext.getResources().getDisplayMetrics();

        float density = dm.density;
        int width = (int) (dm.widthPixels / density);

        /*// 屏幕宽度:屏幕宽度（像素）/屏幕密度
        screenWidth = width / density;  // 屏幕宽度(dp)
        screenHeight = height / density;// 屏幕高度(dp)
        screenDensity = density;*/
        return width;
    }

    public static int getScreenWidthPx() {
        //2、通过Resources获取
        DisplayMetrics dm = AYApplication.appContext.getResources().getDisplayMetrics();
        return dm.widthPixels;
    }

    public static int getScreenHeightDp() {
        //2、通过Resources获取
        DisplayMetrics dm = AYApplication.appContext.getResources().getDisplayMetrics();

        float density = dm.density;
        int height = (int) (dm.heightPixels / density);

        /*// 屏幕宽度:屏幕宽度（像素）/屏幕密度
        screenWidth = width / density;  // 屏幕宽度(dp)
        screenHeight = height / density;// 屏幕高度(dp)
        screenDensity = density;*/
        return height;
    }
}

File: app/src/main/java/com/blackmirror/dongda/utils/DensityUtils.kt
package com.blackmirror.dongda.utils

import com.blackmirror.dongda.base.AYApplication

/**
 * Create By Ruge at 2018-12-06
 */

fun px2dp(pxVal: Float): Float {
    val scale = AYApplication.appContext.resources.displayMetrics.density
    return pxVal / scale
}

/**
 * dp转px
 *
 * @param dp
 * @return
 */
fun dp2px(dp: Int): Int {
    val scale = AYApplication.appContext.resources.displayMetrics.density
    return (dp * scale + 0.5f).toInt()
}

fun getScreenWidthDp(): Int {
    //2、通过Resources获取
    val dm = AYApplication.appContext.resources.displayMetrics

    val density = dm.density

    /*// 屏幕宽度:屏幕宽度（像素）/屏幕密度
            screenWidth = width / density;  // 屏幕宽度(dp)
            screenHeight = height / density;// 屏幕高度(dp)
            screenDensity = density;*/
    return (dm.widthPixels / density).toInt()
}

fun getScreenWidthPx(): Int {
    //2、通过Resources获取
    val dm = AYApplication.appContext.resources.displayMetrics
    return dm.widthPixels
}

fun getScreenHeightDp(): Int {
    //2、通过Resources获取
    val dm = AYApplication.appContext.resources.displayMetrics

    val density = dm.density

    /*// 屏幕宽度:屏幕宽度（像素）/屏幕密度
                screenWidth = width / density;  // 屏幕宽度(dp)
                screenHeight = height / density;// 屏幕高度(dp)
                screenDensity = density;*/
    return (dm.heightPixels / density).toInt()
}